### PR TITLE
Add support for reading IDCODE and testing BYPASS via JTAG

### PR DESCRIPTION
**Pull request by xorptr**
Add support for reading IDCODE and testing BYPASS via JTAG from a device connected to target

### CODE REVIEW

File: examples/htool_jtag.c [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  printf("IDCODE: 0x%08x\n", response.idcode);
+  return 0;
+}
+
+// Parse exactly `expected_bytes_count` byte tokens from `param_value` string
+// into `byte_sequence`.
+// If fewer than `expected_bytes_count` byte tokens are present in
+// `param_value`, return code indicates error.
+// If more than `expected_bytes_count` byte tokens are present in `param_value`,
+// extra bytes are ignored and no error is indicated
```
Comment (xorptr) [inline examples/htool_jtag.c:81]: I am assuming this behavior is okay. If an error needs to be returned, I can change the code here.

File: examples/htool_jtag.c [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      .operation =
+          {
+              .clk_idiv = (uint16_t)clk_idiv,
+              .operation = EC_JTAG_OP_TEST_BYPASS,
+          },
+  };
+
+  int ret = parse_string_param_into_byte_sequence(
+      tdi_bytes, request.params.tdi_pattern, EC_JTAG_TEST_BYPASS_PATTERN_LEN);
+  if (ret < 0) {
```
Comment (xorptr) [inline examples/htool_jtag.c:141]: Change to `ret != 0`

File: examples/htool_jtag.c (hunk 1 above)
Comment (rkr35) [reply examples/htool_jtag.c:81]: Someone trying to send a pattern greater than 64 bytes might be confused why htool is truncating their input, in which case a printed warning would clarify matters. Likewise, someone trying to send exactly 64 bytes should be made aware when their input is unintentionally longer.

File: examples/htool_jtag.c (hunk 2 above)
Comment (rkr35) [reply examples/htool_jtag.c:141]: Agreed

File: examples/htool_jtag.c (hunk 1 above)
Comment (xorptr) [reply examples/htool_jtag.c]: Fixed. The new changes should have the function accept exactly `expected_bytes_count` bytes in form of string tokens (no more and no less)

File: examples/htool_jtag.c (hunk 2 above)
Comment (xorptr) [reply examples/htool_jtag.c]: Fixed

**Review by rkr35 — APPROVED** [review]

**Pull request merged**